core/modules/field/modules/number/number.module

```diff
@@ -122,6 +122,60 @@ function number_field_instance_settings_form_validate($element, &$form_state, $f
       '%min' => $min['#title'],
     )));
   }
+
+  // Consider the value limits of the database, prevent impossible settings.
+  $field_type = $form['#field']['type'];
+  if ($field_type == 'number_integer') {
+    // @see https://mariadb.com/kb/en/int/
+    $min_db_integer = -2147483647;
+    $max_db_integer = 2147483647;
+    $settings = $form_state['values']['instance']['settings'];
+
+    if (!empty($settings['min']) && $settings['min'] < $min_db_integer) {
+      form_error($min, t('%min is smaller than the smallest value the database can save. Choose a value equal to or higher than %db_min.', array(
+        '%min' => $settings['min'],
+        '%db_min' => $min_db_integer,
+      )));
+    }
+    if (!empty($settings['max']) && $settings['max'] > $max_db_integer) {
+      form_error($max, t('%max is greater than the largest value the database can save. Choose a value equal to or lower than %db_max.', array(
+        '%max' => $settings['max'],
+        '%db_max' => $max_db_integer,
+      )));
+    }
+  }
+  elseif ($field_type == 'number_decimal') {
+    // The field settings for precision and scale also dictate the possible
+    // min/max values for the field.
+    $precision = $form['#field']['settings']['precision'];
+    $scale = $form['#field']['settings']['scale'];
+    $settings = $form_state['values']['instance']['settings'];
+
+    if (!empty($settings['min'])) {
+      $min_value = abs($settings['min']);
+      $digits = explode('.', $min_value);
```

Comment:
If this is meant to handle decimal points, then keep in mind that thousands/decimals points are different in some locales (Greek for example uses `.` for thousands and `,` for decimals). Perhaps this is only an issue when using text fields for specifying min/max values, and not when using proper number inputs. Perhaps this is something entirely different than what I think it is.

Reply:
The min, max and step attribute are supposed to use dots, regardless of locale.